node/src/accountant/scanners/scan_schedulers.rs

```diff
@@ -0,0 +1,839 @@
+// Copyright (c) 2025, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::accountant::scanners::StartScanError;
+use crate::accountant::{
+    Accountant, ResponseSkeleton, ScanForNewPayables, ScanForPendingPayables, ScanForReceivables,
+    ScanForRetryPayables,
+};
+use crate::sub_lib::accountant::ScanIntervals;
+use crate::sub_lib::utils::{
+    NotifyHandle, NotifyHandleReal, NotifyLaterHandle, NotifyLaterHandleReal,
+};
+use actix::{Actor, Context, Handler};
+use masq_lib::logger::Logger;
+use masq_lib::messages::ScanType;
+use std::fmt::Debug;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+pub struct ScanSchedulers {
+    pub payable: PayableScanScheduler,
+    pub pending_payable: SimplePeriodicalScanScheduler<ScanForPendingPayables>,
+    pub receivable: SimplePeriodicalScanScheduler<ScanForReceivables>,
+    pub reschedule_on_error_resolver: Box<dyn RescheduleScanOnErrorResolver>,
+    pub automatic_scans_enabled: bool,
+}
+
+impl ScanSchedulers {
+    pub fn new(scan_intervals: ScanIntervals, automatic_scans_enabled: bool) -> Self {
+        Self {
+            payable: PayableScanScheduler::new(scan_intervals.payable_scan_interval),
+            pending_payable: SimplePeriodicalScanScheduler::new(
+                scan_intervals.pending_payable_scan_interval,
+            ),
+            receivable: SimplePeriodicalScanScheduler::new(scan_intervals.receivable_scan_interval),
+            reschedule_on_error_resolver: Box::new(RescheduleScanOnErrorResolverReal::default()),
+            automatic_scans_enabled,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum PayableScanSchedulerError {
+    ScanForNewPayableAlreadyScheduled,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum SchedulingAfterHaltedScan {
+    Schedule(ScanType),
+    DoNotSchedule,
+}
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum HintableScanner {
+    NewPayables,
+    RetryPayables,
+    PendingPayables { initial_pending_payable_scan: bool },
+}
+
+impl From<HintableScanner> for ScanType {
+    fn from(hintable_scanner: HintableScanner) -> Self {
+        match hintable_scanner {
+            HintableScanner::NewPayables => ScanType::Payables,
+            HintableScanner::RetryPayables => ScanType::Payables,
+            HintableScanner::PendingPayables { .. } => ScanType::PendingPayables,
+        }
+    }
+}
+
+pub struct PayableScanScheduler {
+    pub new_payable_notify_later: Box<dyn NotifyLaterHandle<ScanForNewPayables, Accountant>>,
+    pub dyn_interval_computer: Box<dyn NewPayableScanDynIntervalComputer>,
+    pub inner: Arc<Mutex<PayableScanSchedulerInner>>,
+    pub nominal_interval: Duration,
```

Comment:
You'd like to rename this field such that something else is used instead of nominal.